waves_gateway/controller/gateway_controller_impl.py

```diff
@@ -108,10 +115,40 @@ def create_address(self, waves_address: str) -> str:
     def get_attempt_list_by_id(self, attempt_list_id: str) -> Optional[TransactionAttemptList]:
         return self._attempt_list_storage.find_by_attempt_list_id(attempt_list_id)
 
+    @method_doc_inherit
+    def get_block_heights(self):
+        last_checked_coin_block_height = self._key_value_storage.get_last_checked_coin_block_height()
+        last_checked_waves_block_height = self._key_value_storage.get_last_checked_waves_block_height()
+
+        return {
+            "last_checked_coin_block_height": last_checked_coin_block_height,
+            "last_checked_waves_block_height": last_checked_waves_block_height
+        }
+
+    @method_doc_inherit
+    def get_average_attempt_list_tries(self):
+        return self._attempt_list_storage.get_average_attempt_list_tries()
+
+    def get_failed_transactions(self):
+        return self._failed_transaction_storage.get_failed_transactions()
+
+    def get_log_messages(self):
+        return self._log_storage_service.get_messages()
+
     @method_doc_inherit
     def query_attempt_lists(self, query: AttemptListQuery) -> List[TransactionAttemptList]:
         return self._attempt_list_storage.query_attempt_lists(query)
 
     @method_doc_inherit
     def get_attempt_list_by_trigger(self, trigger: AttemptListTrigger) -> Optional[TransactionAttemptList]:
         return self._attempt_list_storage.find_by_trigger(trigger)
+
+    @method_doc_inherit
+    def trigger_attemptlist_retry(self, id: str) -> str:
```

Comment:
Why do you a string return type here? Please use boolean.

Reply:
fixed